### PR TITLE
Nicer URLs

### PR DESCRIPTION
**Pull request by callumacrae**
## Ticket and context

Replacing https://github.com/DFE-Digital/ecf-engage-and-learn/pull/172, which broke

Ticket: https://dfedigital.atlassian.net/browse/CPDEL-571

### Is there anything weird that code reviewer should know?
not really

### Review Checks
- [x] All commit messages ~~are meaningful and true~~ will be squashed
- [x] Added enough automated tests
- [x] All forms have an `id` attribute on them

### How can someone see it it review app?
1. Click the link to review app (posted by a `github-actions` bot below)
2. Follow the steps from the ticket.

### CODE REVIEW

File: app/helpers/load_resources_helper.rb [hunk 1]
```diff
@@ -0,0 +1,103 @@
+# frozen_string_literal: true
```
Comment (callumacrae) [inline app/helpers/load_resources_helper.rb:1]: Note for reviewers: this file isn't tested using helper tests. Given that it's testing params and using instance variables, I think it's easier to test using request tests. Feel free to disagree!

File: spec/requests/core_induction_programme/course_year_spec.rb [hunk 2]
```diff
@@ -22,15 +22,15 @@
...
     describe "GET /years/new" do
       it "renders the cip new years page" do
-        get "/years/new"
+        get "/#{cip.to_param}/new"
         expect(response).to render_template(:new)
       end
     end
 
-    describe "POST /years" do
+    xdescribe "POST /years" do
```
Comment (callumacrae) [inline spec/requests/core_induction_programme/course_year_spec.rb:30]: the `xdescribe`s in this file are intentional - i'm not sure why the tests used to work, but the functionality is broken when access through a browser so they shouldn't have been. needs fixing in a separate PR (and these tests bringing back with it)

File: spec/factories/core_induction_programmes.rb [hunk 3]
```diff
@@ -3,10 +3,15 @@
...
+    sequence(:slug) { |n| "test-cip-#{n}" }
 
     trait :with_course_year do
       course_year_one_id { FactoryBot.create(:course_year).id }
       course_year_two_id { FactoryBot.create(:course_year).id }
     end
+
+    after :create do
+      Rails.application.reload_routes!
+    end
```
Comment (kicferk1) [inline spec/factories/core_induction_programmes.rb:15]: we don't need it anymore, right?

File: spec/requests/core_induction_programme/course_year_spec.rb (hunk 2 above)
Comment (kicferk1) [reply spec/requests/core_induction_programme/course_year_spec.rb:30]: Could you make a JIRA ticket for fixing it and leave a code comment linking to it?

**Review by kicferk1 — COMMENTED** [review]
Looks pretty damn good, a few small questions.

File: spec/requests/core_induction_programme/course_year_spec.rb (hunk 2 above)
Comment (callumacrae) [reply spec/requests/core_induction_programme/course_year_spec.rb:30]: done! https://dfedigital.atlassian.net/browse/CPDEL-649

**Review by kicferk1 — APPROVED** [review]
Let's not merge it just yet - I want to release it to staging tomorrow morning, so I can see if there are any obvious issues, before we push it out to prod. And I want to push some prod fixes this afternoon :)

**Comment by kicferk1** [comment]
Also, some conflicts (sorryyyy)

**Comment by callumacrae** [comment]
> Also, some conflicts (sorryyyy)

I'll fix them as soon to merging as possible!

**Pull request merged**